### PR TITLE
Added profiling documentation

### PR DESCRIPTION
**Pull request by kindermoumoute**
This doc explains how to profile snap and exploit the results with go-torch and pprof.

@intelsdi-x/snap-maintainers

### CODE REVIEW

**Comment by danielscottt** [comment]
@kindermoumoute hey man, looks like you need to rebase master in to resolve some conflicts.  While you're at it, would you also mind fixing these commits up into a single commit?

**Comment by lynxbat** [comment]
Could we also add a series of current profiling svg's to this? Or is that already the plan?

**Comment by kindermoumoute** [comment]
@lynxbat I'm writing a blog post on medium about the results I got!

**Comment by mjbrender** [comment]
@kindermoumoute awesome idea! Let me or @sarahjhh know when your post is in draft and we'll happily take a look (then publish here: https://medium.com/intel-sdi). Thanks 👍

File: docs/PROFILING.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+Using pprof and go-torch is a good way to bring to light on how your application performs. If you've never heard about profiling with the Go programming language, feel free to take a look at [Debugging performance issues in Go programs](https://software.intel.com/en-us/blogs/2014/05/10/debugging-performance-issues-in-go-programs).
+
+## Set up ...
+### FlameGraph
+Flame graphs are a visualization of profiled software, allowing the most frequent code-paths to be identified quickly and accurately. FlameGraph needs to be installed with Perl before trying to use go-torch.
+
+You can install Perl with your program manager. The following example uses `brew` on OSX:    
+```bash
+brew install perl
```
Comment (IRCody) [inline docs/PROFILING.md]: Can we also add Apt/yum install commands here.

File: docs/PROFILING.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+```
+
+Then download flamegraph.pl from the [flamegraph repository](https://github.com/brendangregg/FlameGraph) and move it into your preferred $PATH directory, e.g. `~/bin`:
+```bash
+mv flamegraph.pl ~/bin/flamegraph
```
Comment (IRCody) [inline docs/PROFILING.md]: Add wget and chmod to make it executable.

File: docs/PROFILING.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+```
+(Don't forget to remove the file extension.)
+
+### graphviz
+Graphviz will allow you to generate graphs with pprof.
+```
+brew install graphviz
```
Comment (IRCody) [inline docs/PROFILING.md]: Also add yum/apt here?

**Comment by IRCody** [comment]
@kindermoumoute: have you had a chance to update this?

**Comment by IRCody** [comment]
LGTM

**Pull request merged**